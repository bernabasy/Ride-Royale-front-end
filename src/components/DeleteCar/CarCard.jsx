/* eslint-disable camelcase */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import styles from './style.css';
const api_url = 'http://127.0.0.1:3000/api/v1/cars/';

const DeleteCarUi = ({
  id, picture, model, cars, setCars,
}) => {
  const [errorMessage, setErrorMessage] = useState('');
  const removeCar = (id) => {
    axios
      .delete(`${api_url}${id}/delete`)
      .then(() => {
        let filteredCars = [];
        filteredCars = cars.filter((car) => car.id !== id);
        setCars(filteredCars);
      })
      .catch((error) => {
        setErrorMessage(error.response.data.error);
      });
  };

  return (
    <>
      <p className={`${styles.errorMsg} fs-5`}>{errorMessage}</p>
      <section className={`${styles.carCnt} mx-4`} key={id}>
        <div className={styles.imgCnt}>
          <img src={picture} alt={model} className={styles.carImg} />
        </div>
        <div className="model-detail d-flex flex-column justify-content-center align-items-center">
          <h3 className="h4 mt-3">{model}</h3>
        </div>
        <button
          type="button"
          onClick={() => removeCar(id)}
          className="btn btn-danger mb-3"
        >
          Delete
        </button>
      </section>
    </>
  );
};

DeleteCarUi.propTypes = {
  picture: PropTypes.string.isRequired,
  model: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  cars: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.object])).isRequired,
  setCars: PropTypes.func.isRequired,
};

export default DeleteCarUi;