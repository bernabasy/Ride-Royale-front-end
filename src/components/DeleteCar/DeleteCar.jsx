import React, { useState, useEffect } from 'react';
import axios from 'axios';
import { useSelector } from 'react-redux';
import CarCard from './CarCard';

const DeleteCar = () => {
  const [cars, setCars] = useState([]);
  const [loading, setLoading] = useState(false);
  const user = useSelector((state) => state.user);
  useEffect(() => {
    setLoading(true);
    axios
      .get(`https://luxy-rides.onrender.com/api/v1/cars/${user.user.id}`)
      .then((response) => {
        setCars(response.data.cars);
        setLoading(false);
      });
  }, []);

  if (cars.length <= 0) {
    if (loading) {
      return (
        <div className="container-fluid d-flex justify-content-center align-items-center vh-100">
          <i className="fa-solid fa-spinner fa-spin fs-1" />
        </div>
      );
    }
    return (
      <h2 className="mt-5 text-center">
        You don&apos;t have any cars in our system yet.
      </h2>
    );
  }
  return (
    <>
      <h2 className="h2 text-center mt-5 text-uppercase">
        <strong>WELCOME TO THE DELETE AREA</strong>
      </h2>
      <p className="text-center text-muted mb-5">
        Please click on delete button below to delete a car
      </p>
      <div className="container-fluid d-flex flex-wrap justify-content-center">
        {cars.map((car) => (
          <CarCard
            key={car.id}
            id={car.id}
            image={car.picture}
            model={car.model}
            driver_name={car.driver_name}
            cars={cars}
            setCars={setCars}
          />
        ))}
      </div>
    </>
  );
};
export default DeleteCar;