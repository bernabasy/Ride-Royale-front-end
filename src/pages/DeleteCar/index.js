import React from 'react';
import Layout from '../../components/Layout/Layout';
import DeleteCar from '../../components/DeleteCar/DeleteCar';

const DeleteCarItems = () => (
  <Layout>
    <DeleteCar />
  </Layout>
);

export default DeleteCarItems;
