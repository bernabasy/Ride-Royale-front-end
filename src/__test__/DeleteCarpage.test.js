import renderer from 'react-test-renderer';
import DeleteCar from '../components/DeleteCar/DeleteCar';

it('Render the DeleteCar component', () => {
  const tree = renderer.create(
    <DeleteCar />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
