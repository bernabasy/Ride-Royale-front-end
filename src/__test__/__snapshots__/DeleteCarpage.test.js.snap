// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the DeleteCar component 1`] = `
<h2
  className="mt-5 text-center"
>
  You don't have any cars in our system yet.
</h2>
`;
